contracts/Strategy.sol

```diff
@@ -274,60 +371,317 @@ contract Strategy is BaseStrategy {
+                    pairTotalAssets
+                );
+            }
+        }
+
+        for (uint256 i = 0; i < kashiPairs.length; i++) {
+            if (kashiPairsIncreasedAllocation[i] == 0) continue;
+
+            uint256 sharesInBento = sharesInBento();
+            uint256 sharesToAdd =
+                wantToBentoShares(kashiPairsIncreasedAllocation[i], true);
+
+            if (sharesToAdd > sharesInBento) {
+                sharesToAdd = sharesInBento;
+            }
+
+            depositInKashiPair(i, sharesToAdd);
+        }
+    }
+
+    function depositInKashiPair(uint256 kashiPairIndex, uint256 sharesToDeposit)
+        internal
+    {
+        bentoBox.transfer(
+            BIERC20(address(want)),
+            address(this),
+            address(kashiPairs[kashiPairIndex].kashiPair),
+            sharesToDeposit
+        );
+
+        uint256 depositedFraction =
+            kashiPairs[kashiPairIndex].kashiPair.addAsset(
+                address(this),
+                true,
+                sharesToDeposit
+            );
+
+        if (kashiPairs[kashiPairIndex].pid != 0) {
+            masterChef.deposit(
+                kashiPairs[kashiPairIndex].pid,
+                depositedFraction
+            );
+        }
+    }
+
+    function liquidateKashiPair(uint256 kashiPairIndex, uint256 sharesToFree)
+        internal
+        returns (uint256 _shareLiquiduated)
+    {
+        (, uint256 lastAccrued, ) =
+            kashiPairs[kashiPairIndex].kashiPair.accrueInfo();
+        if (block.timestamp > lastAccrued) {
+            // We need to call accrue to accurately calculate totalAssets
+            kashiPairs[kashiPairIndex].kashiPair.accrue();
+        }
+
+        uint256 liquidShares = kashiPairLiquidShares(kashiPairIndex);
+        if (sharesToFree > liquidShares) {
+            sharesToFree = liquidShares;
+        }
+
+        uint256 fractionsToFree =
+            bentoSharesToKashiFraction(
+                kashiPairs[kashiPairIndex].kashiPair,
+                sharesToFree,
+                true
             );
 
-            kashiPair.addAsset(address(this), true, sharesInBento);
+        // Remove from masterChef if there is a non-zero pid
+        if (kashiPairs[kashiPairIndex].pid != 0) {
+            uint256 fractionInMc = kashiFactionInMasterChef(kashiPairIndex);
+            uint256 fractionsToFreeFromMc = fractionsToFree;
+            if (fractionsToFreeFromMc > fractionInMc) {
+                fractionsToFreeFromMc = fractionInMc;
+            }
+            masterChef.withdraw(
+                kashiPairs[kashiPairIndex].pid,
+                fractionsToFreeFromMc
+            );
+        }
+
+        uint256 fractionBalance = kashiFractionInPair(kashiPairIndex);
+
+        if (fractionsToFree > fractionBalance) {
+            fractionsToFree = fractionBalance;
+        }
+
+        _shareLiquiduated = kashiPairs[kashiPairIndex].kashiPair.removeAsset(
```

Comment:
_shareLiquiduated -> _shareLiquidated

small typo that might lead to issues in the future